fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1167,6 +1177,63 @@ void testDataLoad() {
         context.close();
     }
 
+    @ParameterizedTest
+    @BooleanSource
+    void testSimpleUpdate(boolean primaryKeySegmentIndexEnabled) {
+        final Index index = primaryKeySegmentIndexEnabled ? SIMPLE_TEXT_SUFFIXES_WITH_PRIMARY_KEY_SEGMENT_INDEX : SIMPLE_TEXT_SUFFIXES;
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_MERGE_SEGMENTS_PER_TIER, 3.0)
+                .build();
+        final String[] nums = { "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine" };
+        for (int i = 0; i < 20; i++) {
+            final int ii = i + 1;
+            try (FDBRecordContext context = openContext(contextProps)) {
+                rebuildIndexMetaData(context, SIMPLE_DOC, index);
+                final String factors = IntStream.range(1, nums.length)
+                        .filter(n -> ii % n == 0)
+                        .mapToObj(n -> nums[n])
+                        .collect(Collectors.joining(" "));
+                recordStore.saveRecord(createSimpleDocument(1000L + i % 5, factors, null),
+                        i < 5 ? FDBRecordStoreBase.RecordExistenceCheck.ERROR_IF_EXISTS : FDBRecordStoreBase.RecordExistenceCheck.ERROR_IF_NOT_EXISTS);
+                context.commit();
+            }
+        }
+
+        if (primaryKeySegmentIndexEnabled) {
+            assertThat(timer.getCount(LuceneEvents.Counts.LUCENE_MERGE_SEGMENTS), greaterThan(10));
+            assertThat(timer.getCount(LuceneEvents.Counts.LUCENE_DELETE_DOCUMENT_BY_QUERY), equalTo(0));
+            assertThat(timer.getCount(LuceneEvents.Counts.LUCENE_DELETE_DOCUMENT_BY_PRIMARY_KEY), greaterThan(10));
+        } else {
+            assertThat(timer.getCount(LuceneEvents.Counts.LUCENE_DELETE_DOCUMENT_BY_QUERY), greaterThan(10));
+            assertThat(timer.getCount(LuceneEvents.Counts.LUCENE_DELETE_DOCUMENT_BY_PRIMARY_KEY), equalTo(0));
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            assertIndexEntryPrimaryKeys(List.of(1002L), // 18
+                    recordStore.scanIndex(index, fullTextSearch(index, "three"), null, ScanProperties.FORWARD_SCAN));
+            assertIndexEntryPrimaryKeys(List.of(1000L, 1004L),  // 16,20
+                    recordStore.scanIndex(index, fullTextSearch(index, "four"), null, ScanProperties.FORWARD_SCAN));
+            assertIndexEntryPrimaryKeys(List.of(),
+                    recordStore.scanIndex(index, fullTextSearch(index, "seven"), null, ScanProperties.FORWARD_SCAN));
```

Comment:
It seems like it would also be good to do some deletes and test that, but as you noted in a TODO, if a segment runs out of live documents, this code doesn't currently clean them up.

Reply:
Also, it would be good to have a test that deletes and recreates a new record with the same primary key.

Reply:
Added a delete test.